taxis/views.py

```diff
@@ -1,3 +1,80 @@
-from django.shortcuts import render
+import json
+import datetime
 
-# Create your views here.
+from django.http.response   import JsonResponse
+from django.views           import View
+from django.db.models       import Q
+
+from taxis.models           import *
+class TaxiListView(View):
+    def get(self, request):
+        departure_location_name  = request.GET.get('departure_location_name', None)
+        arrival_location_name    = request.GET.get('arrival_location_name', None)
+        seat_type                = request.GET.get('seat_type')
+        seat_remain              = request.GET.get('seat_remain')
+        departure_date_string    = request.GET.get('departure_date')
+        
+        departure_time           = request.GET.get('departure_time', '1900-01-01 23:59')
+        price                    = request.GET.get('price', 60000)
+        taxi_company             = request.GET.getlist('taxi_company', ['Dasul Taxi', 'Taxi Choi-gging', 'Art Transportation', 'Lama 운수', 'DaMo taxi', 'Muy bien Trans'])
+        sort                     = request.GET.get('sort', None)
+
+        sort_list = {
+            'dep_time': '-course__departure_time',
+            'price'   : '-price',
+            None      : '-price'
+        }
+        sort           = None if sort not in sort_list else sort
+        sort_string    = sort_list[sort]
```

Comment:
```suggestion
        sort                     = request.GET.get('sort',  '-price')

        sort_list = {
            'dep_time': '-course__departure_time',
            'price'   : '-price'
        }
        sort_string    = sort_list[sort]
```
None 상황까지 잘 생각해 주셨습니다.!!! goooooood~~~
GET.get()을 사용 할 때 기본 값을 설정 할 수 있습니다. 그러면 key가 없을 때 None이 아닌 내가 설정한 값을 받을 수 있습니다. 위에 제가 제시한 코드 보시고 수정 부탁드립니다.

Reply:
영건님의 작품에 숟가락을 얹었습니당 ㅋㅋㅋㅋ none값으로 -price를 넣어주니 한줄이 삭제돼서 더 깔끔해진 것 같네요!! 감사합니당